x/profile/types/genesis.go

```diff
@@ -37,10 +37,15 @@ func (gs GenesisState) Validate() error {
 
 	// Check for duplicated ID in coordinator
 	coordinatorIDMap := make(map[uint64]bool)
+	count := len(gs.CoordinatorList)
```

Comment:
We must check with the `coordinatorCount` value in `genesisState` because coordinator can be removed and it will not be taken into account if we do `len(gs.CoordinatorList)`

"count" can be actually a bit ambiguous, this is the count of created coordinators so far and not the actual count of current coordinators